### DISCUSSION TITLE
-depth 8 not converting

### DISCUSSION THREAD

**Discussion by Vanir-Labs** [ImageMagick/ImageMagick#6608] [Help] [answered]
### ImageMagick version

7.1.1.15 Q16-HDRI 64x

### Operating system, version and so on

Windows 11 64x

### Description

Does anyone know why convert and mogrify are not changing the bit depth to 8 with these arguments?

```
convert "C:\Users\rivers\image\image32d.bmp" -depth 8 "C:\Users\rivers\image\image8d.bmp"
```

**Comment by snibgo** [comment]
`-depth 8` is the number of bits per channel per pixel. It isn't the total number of bits per pixel. If there are 3 channels, the total bits per pixel will be 24.

BMP files have a maximum of 8 bits per channel per pixel.

If you want a total of 8 bits per pixel, and 3 channels, I suppose you want a palette format.
```
magick in.bmp -type palette out.bmp
```

**Reply by Vanir-Labs** [reply]
Whenever I process any of the commands (magick, convert, mogrify) it changes my picture from 24b (or 8b) to 32b depth.
![image](https://github.com/ImageMagick/ImageMagick/assets/82656075/ec941c57-3ce4-4a27-a8b3-c41abb4c77fe)

Am I just not understanding things or is there something not correct here?

**Comment by snibgo** [comment, marked as answer]
32 bits/pixel would be for 4 channels, including alpha. Use `-alpha off` before the output filename to remove alpha.

`magick` for me works fine, without adding an alpha channel.

I encourage you not to use `convert` or `mogrify`. Instead, use `magick` or `magick mogrify`.

**Reply by Vanir-Labs** [reply]
My dude. This was it. Many many thanks for this. You really helped my ass out.
![image](https://github.com/ImageMagick/ImageMagick/assets/82656075/14714d43-a2c5-41f9-b182-e7256d5add2f)

**Comment by Vanir-Labs** [comment]
magick in.bmp -type palette -alpha disassociate out.bmp
